src/apply.jl

```diff
@@ -87,52 +86,124 @@ end
 function ITensors.apply(
   o⃗::Vector{ITensor},
   ψ::AbstractITensorNetwork;
-  cutoff,
-  maxdim=typemax(Int),
   normalize=false,
   ortho=false,
-  kwargs...,
+  apply_kwargs...,
 )
   o⃗ψ = ψ
   for oᵢ in o⃗
-    o⃗ψ = apply(oᵢ, o⃗ψ; cutoff, maxdim, normalize, ortho)
+    o⃗ψ = apply(oᵢ, o⃗ψ; normalize, ortho, apply_kwargs...)
   end
   return o⃗ψ
 end
 
 function ITensors.apply(
-  o⃗::Scaled,
-  ψ::AbstractITensorNetwork;
-  cutoff,
-  maxdim,
-  normalize=false,
-  ortho=false,
-  kwargs...,
+  o⃗::Scaled, ψ::AbstractITensorNetwork; normalize=false, ortho=false, apply_kwargs...
 )
   return maybe_real(Ops.coefficient(o⃗)) *
-         apply(Ops.argument(o⃗), ψ; cutoff, maxdim, normalize, ortho, kwargs...)
+         apply(Ops.argument(o⃗), ψ; cutoff, maxdim, normalize, ortho, apply_kwargs...)
 end
 
 function ITensors.apply(
-  o⃗::Prod,
-  ψ::AbstractITensorNetwork;
-  cutoff,
-  maxdim,
-  normalize=false,
-  ortho=false,
-  kwargs...,
+  o⃗::Prod, ψ::AbstractITensorNetwork; normalize=false, ortho=false, apply_kwargs...
 )
   o⃗ψ = ψ
   for oᵢ in o⃗
-    o⃗ψ = apply(oᵢ, o⃗ψ; cutoff, maxdim, normalize, ortho, kwargs...)
+    o⃗ψ = apply(oᵢ, o⃗ψ; normalize, ortho, apply_kwargs...)
   end
   return o⃗ψ
 end
 
 function ITensors.apply(
-  o::Op, ψ::AbstractITensorNetwork; cutoff, maxdim, normalize=false, ortho=false, kwargs...
+  o::Op, ψ::AbstractITensorNetwork; normalize=false, ortho=false, apply_kwargs...
 )
-  return apply(ITensor(o, siteinds(ψ)), ψ; cutoff, maxdim, normalize, ortho, kwargs...)
+  return apply(ITensor(o, siteinds(ψ)), ψ; normalize, ortho, apply_kwargs...)
+end
+
+#In the future we will try to unify this into apply() above but currently leave it mostly as a separate function
+"""Apply() function for an ITN in the Vidal Gauge. Hence the bond tensors are required.
+Gate does not necessarily need to be passed. Can supply an edge to do an identity update instead. Uses Simple Update procedure assuming gate is two-site"""
+function ITensors.apply(
+  o::Union{ITensor,NamedEdge},
+  ψ::AbstractITensorNetwork,
+  bond_tensors::DataGraph;
+  normalize=false,
+  apply_kwargs...,
+)
+  ψ = copy(ψ)
+  bond_tensors = copy(bond_tensors)
+  v⃗ = typeof(o) == ITensor ? neighbor_vertices(ψ, o) : [src(o), dst(o)]
```

Comment:
I try to avoid code like this that uses `if` statements based on checking types. I find it makes the code cleaner to instead define a function and then dispatch on that function.

So for example:
```julia
_gate_vertices(o::ITensor, ψ) = neighbor_vertices(ψ, o)
_gate_vertices(o::AbstractEdge, ψ) = [src(o), dst(o)]
```
and then use:
```julia
  v⃗ = _gate_vertices(o, ψ)
```